pkg/workloads/cortex/lib/storage/concurrency.py

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+# Copyright 2020 Cortex Labs, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os, fcntl, time
+
+
+class FileLock:
+    def __init__(self, lock_file, timeout=None):
+        """
+        lock_file - File to use as lock.
+        timeout - If used, a timeout exception will be raised if the lock can't be acquired. Measured in seconds.
+        """
+        self._lock_file = lock_file
+        self._file_handle = None
+
+        self.timeout = timeout
+        self._time_loop = 0.001
+
+        # create lock if it doesn't exist
+        try:
+            with open(self._lock_file_handlefile, "w+") as f:
+                pass
+        except:
+            pass
```

Comment:
Is this a recoverable error, or should we let it propagate rather than catching it?

Reply:
Okay, that can actually be removed and it's better this way because the `with` context manager might throw, for instance, a permission denied error, in which case it's just better to let the app exit with a non-zero code.